packages/harmony/src/components/text/Text.tsx

```diff
@@ -42,15 +42,18 @@ export const Text = forwardRef(
       position: 'relative',
       boxSizing: 'border-box',
       ...(color &&
-        color === 'heading' && {
-          // inline is necessary to prevent text clipping
-          display: 'inline',
-          color: theme.color.secondary.secondary,
-          WebkitTextFillColor: 'transparent',
-          backgroundClip: 'text',
-          backgroundImage: theme.color.text.heading
-        }),
-      ...(color && color !== 'heading' && { color: theme.color.text[color] }),
+        (color === 'heading'
+          ? {
+              // inline is necessary to prevent text clipping
+              display: 'inline',
+              color: theme.color.secondary.secondary,
+              WebkitTextFillColor: 'transparent',
+              backgroundClip: 'text',
+              backgroundImage: theme.color.text.heading
+            }
+          : color === 'inherit'
+          ? { color: 'inherit' }
+          : { color: theme.color.text[color] })),
```

Comment:
can we break this out, it's kinda hard to read

Reply:
yeah agreed, on it